src/LetsEncrypt.Azure.Core.V2/DnsLookupService.cs

```diff
@@ -19,56 +20,45 @@ public DnsLookupService(ILogger<DnsLookupService> logger = null)
             this.logger = logger ?? NullLogger<DnsLookupService>.Instance;
         }
 
-        public async Task<bool> Exists(string hostname, string dnsTxt, int timeout = 60)
+        public async Task<bool> Exists(string zoneName, string dnsTxt, int timeout = 60)
         {
-            logger.LogInformation("Starting dns precheck validation for hostname: {HostName} challenge: {Challenge} and timeout {Timeout}", hostname, dnsTxt, timeout);
+            logger.LogInformation("Starting dns precheck validation for hostname: {HostName} challenge: {Challenge} and timeout {Timeout}", zoneName, dnsTxt, timeout);
             var idn = new IdnMapping();
-            hostname = idn.GetAscii(GetNoneWildcardDomain(hostname));
-            var dnsClient = GetDnsClient(hostname);
+            zoneName = idn.GetAscii(zoneName);
+            var dnsClient = GetDnsClient(zoneName);
             var startTime = DateTime.UtcNow;
-            string queriedDns = "";
-            //Lets encrypt checks a random authoritative server, thus we need to ensure that all respond with the challenge. 
-            foreach (var ns in dnsClient.NameServers)
+            bool result = false;
+            do
             {
-                logger.LogInformation("Validating dns challenge exists on name server {NameServer}", ns.ToString());
-                do
+                var servers = dnsClient.NameServers.Select(s => s.Endpoint.Address).ToImmutableArray();
+                logger.LogInformation("Validating dns challenge exists on name servers {NameServers}", string.Join(", ", servers));
+                var dnsRes = dnsClient.QueryServer(servers, $"_acme-challenge.{zoneName}", QueryType.TXT);
+                result = dnsRes.Answers.TxtRecords().FirstOrDefault()?.Text.Any(r => r == dnsTxt) ?? false;
```

Comment:
It is not enough that one of the Nameserver have the right TXT record, because Lets encrypt server pick one at random. So I think they should be a check that all NS servers have the record.